encord/transformers/coco/coco_encoder.py

```diff
@@ -612,8 +705,168 @@ def get_skeleton(self, object_: dict, image_id: int, size: Size) -> Union[CocoAn
+                safe_dict_update(res, self.get_checklist_answer(attribute, answers))
+
+        self.add_unselected_attributes(object_feature_hash, res, dynamic=False)
+
+        return res
+
+    def get_id_and_object_hash_to_answers_map(self, object_actions: dict) -> Dict[Tuple[int, str], dict]:
+        if self._id_and_object_hash_to_answers_map is not None:
+            return self._id_and_object_hash_to_answers_map
+
+        res = defaultdict(dict)
+        feature_hash_to_attribute_map = self.get_feature_hash_to_flat_object_attribute_map()
+        for object_hash, payload in object_actions.items():
+            for action in payload["actions"]:
+                feature_hash = action["featureHash"]
+                if feature_hash not in feature_hash_to_attribute_map:
+                    # This will be a deeply nested attribute
+                    continue
+
+                attribute = feature_hash_to_attribute_map[feature_hash]
+                answers = action["answers"]
+                answers_dict = {}
+                if attribute.get_property_type() == PropertyType.TEXT:
+                    safe_dict_update(answers_dict, self.get_text_answer(attribute, answers))
+                elif attribute.get_property_type() == PropertyType.RADIO:
+                    safe_dict_update(answers_dict, self.get_radio_answer(attribute, answers))
+                elif attribute.get_property_type() == PropertyType.CHECKLIST:
+                    safe_dict_update(answers_dict, self.get_checklist_answer(attribute, answers))
+
+                for sub_range in action["range"]:
+                    for i in range(sub_range[0], sub_range[1] + 1):
+                        safe_dict_update(res[(i, object_hash)], answers_dict)
+
+        self._id_and_object_hash_to_answers_map = res
+        return res
+
+    def get_flat_dynamic_classifications(
+        self,
+        object_hash: str,
+        feature_hash: str,
+        image_id: int,
+        id_and_object_hash_to_answers_map: Dict[Tuple[int, str], dict],
+    ) -> dict:
+
+        res = {}
+        id_and_object_hash = (image_id, object_hash)
+        if id_and_object_hash in id_and_object_hash_to_answers_map:
+            res = id_and_object_hash_to_answers_map[(image_id, object_hash)]
+
+        self.add_unselected_attributes(feature_hash, res, dynamic=True)
+
+        return res
+
+    def add_unselected_attributes(self, feature_hash: str, attributes_dict: dict, dynamic: bool) -> None:
+        """
+        Attributes which have never been selected will not show up in the actions map. They will need to be
+        added separately. NOTE: this assumes uniqueness of features. Quite an edge case but if it ever comes
+        up it needs to be solved somewhere here.
+        """
+
+        all_attributes = self.get_attributes_for_feature_hash(feature_hash)
+        for attribute in all_attributes:
+            if attribute.dynamic is dynamic:
```

Comment:
This naming is unfortunate. If the parameter is `False` then this select the "static" attributes, but the code totally doesn't read this way.

Maybe `dynamic_filter` or smth? Don't have a properly constructive suggestion here...